frame/contracts/src/benchmarking/mod.rs

```diff
@@ -896,35 +902,43 @@ benchmarks! {
 	}: call(origin, instance.addr, 0u32.into(), Weight::MAX, None, vec![])
 
 	// Only the overhead of calling the function itself with minimal arguments.
-	// The contract is a bit more complex because I needs to use different keys in order
+	// The contract is a bit more complex because it needs to use different keys in order
 	// to generate unique storage accesses. However, it is still dominated by the storage
-	// accesses.
+	// accesses. We store all the keys that we are about to write at beforehand
+	// because re-writing at an existing key is always more expensive than writing
+	// it at a virgin key.
 	#[skip_meta]
 	seal_set_storage {
 		let r in 0 .. API_BENCHMARK_BATCHES;
-		let keys = (0 .. r * API_BENCHMARK_BATCH_SIZE)
-			.map(|n| T::Hashing::hash_of(&n).as_ref().to_vec())
-			.collect::<Vec<_>>();
-		let key_len = keys.get(0).map(|i| i.len() as u32).unwrap_or(0);
+		let max_key_len = T::MaxStorageKeyLen::get();
+		let keys = (0 .. r * API_BENCHMARK_BATCH_SIZE/2)
```

Comment:
I think you cannot just do that because the weight calculation in `schedule.rs` depends on that. Check the definition of the `cost_batched_args` macro there.